libraries/recipe_helpers.rb

```diff
@@ -195,15 +194,15 @@ def must_reinstall?(node)
 
         include Chef::Mixin::ShellOut
         def agent_get_version
-          return nil unless File.exist?(WIN_BIN_PATH)
+          return nil unless File.file?(WIN_BIN_PATH)
           shell_out("\"#{WIN_BIN_PATH}\" version -n").stdout.strip
         end
 
         def fetch_current_version
           raw_version = agent_get_version
           return nil if raw_version.nil?
           match_data = raw_version.match(/^Agent ([^\s]*) (- Meta: ([^\s]*) )?- Commit/)
-
+          # will fail if raw_version is empty (Line 199 fails)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏We shouldn't mention a line as the file keeps changing !

Reply:
will edit the comment 👍